protocol/contracts/templegold/TempleGoldVesting.sol

```diff
@@ -0,0 +1,339 @@
+        totalVestedAndUnclaimed -= unreleased;
+        _schedule.revoked = true;
+        if(!_vestingIds.remove(_vestingId)) { revert CommonEventsAndErrors.InvalidParam(); }
+        emit Revoked(_vestingId, _schedule.recipient, unreleased, totalVestedAndUnclaimed);
+    }
+
+    /**
+     * @notice Recover ERC20 token
+     * @dev function visibility made public to match parent function
+     * @param _token Token address
+     * @param _to Recipient address
+     * @param _amount Amount to recover
+     */
+    function recoverToken(
+        address _token,
+        address _to,
+        uint256 _amount
+    ) public override onlyElevatedAccess {
+        super.recoverToken(_token, _to, _amount);
+    }
+
+    /**
+     * @notice Set payment token for fixed and epoch payments
+     * @dev function visibility made public to match parent function. ignore compile warning saying visibility can be set to `pure`
+     */
+    function setPaymentToken(address /*_token*/) public pure override {
+        revert NotImplemented();
+    }
+
+    /**
+     * @dev Computes the next vesting schedule identifier for a given account address.
+     */
+    function computeNextVestingScheduleIdForHolder(
+        address _recipient
+    ) public view override returns (bytes32) {
+        return
+            computeVestingScheduleIdForAddressAndIndex(
+                _recipient,
+                holdersVestingCount[_recipient]
+            );
+    }
+
+    /**
+     * @notice Get vesting Id at index from list of vesting Ids
+     * @param _index Index
+     * @return id Bytes32 Id
+     */
+    function getVestingIdAtIndex(
+        uint256 _index
+    ) external view override returns (bytes32 id) {
+        /// @dev revert with an array out-of-bounds access if only `id = _vestingIds.at(_index);` is used
+        if (_vestingIds.length() == 0) { return id; }
+        id = _vestingIds.at(_index);
+    }
+
+    /**
+     * @notice Release tokens vested at current block timestamp. Caller must be recipient of vest
+     * @param _vestingId Vesting Id
+     */
+    function release(bytes32 _vestingId) external override {
+        if(!vestingIdExists(_vestingId)) { revert CommonEventsAndErrors.InvalidParam(); }
+        VestingSchedule storage _schedule = schedules[_vestingId];
+        uint256 _releasableAmount  = _validateRelease(_schedule);
+        _release(_schedule, _releasableAmount);
+        emit Released(_vestingId, msg.sender, _releasableAmount);
+    }
+
+    /**
+     * @dev Computes the vesting schedule identifier for an address and an index.
+     */
+    function computeVestingScheduleIdForAddressAndIndex(
+        address holder,
+        uint256 index
+    ) public pure override returns (bytes32) {
+        return keccak256(abi.encodePacked(holder, index));
+    }
+
+    /**
+     * @notice Check if vesting Id exists
+     * @param _id Vesting Id
+     * @return Bool
+     */
+     function vestingIdExists(bytes32 _id) public view returns (bool) {
```

Comment:
A better name might be `isActiveVestingId(_id)`, since revoked id's still 'exist' in the schedules map but it's just no longer active because it got revoked